docs/complex-types/index.md

```diff
@@ -0,0 +1,21 @@
+# Deep Dive into Complex type within lib
+
+## Exploring Complex TypeScript Types
+
+Welcome to the section where we unravel the intricacies of TypeScript types within our library. As we navigate through the TypeScript landscape, you'll encounter some sophisticated type definitions that power the core of our functionality.
+
+## Understanding TypeScript Types Made Easy
+
+Dive into the inner workings of our library as we break down complex TypeScript types into digestible pieces. We'll guide you through the rationale behind each type, offering clarity on how they contribute to the overall robustness of our code.
+
+## Your Roadmap to TypeScript Mastery
+
+Consider this section your roadmap to mastering TypeScript within the context of our library. We'll introduce you to essential type concepts and demonstrate how they enhance the predictability and reliability of your code.
+
+## Demystifying TypeScript Jargon
+
+If you've ever felt lost in the sea of TypeScript jargon, fear not. We're here to demystify terms, explain intricate type patterns, and provide real-world examples to solidify your understanding.
```

Comment:
To nie jest cel naszej dokumentacji, takie cel mogą mieć ewentualne artykuły na naszym blogu - ale to kwestia przyszłego roku ;)